### PR TITLE
Require all 'checkbox agreement' elements to be checked

### LINKED ISSUE #105: Irregularities with the 'checkbox agreement' question mode in CTRL
For the first two pages of UDN-Aus I need the participant to tick every box before they are able to move on to the next page. I thought this would work by making every question a 'checkbox agreement' and then implementing a module popup that should appear unless every question is ticked. When testing the pop-up appearance is quite irregular and doesn't appear even though some of the boxes aren't checked. Participants are required to tick that they've agreed to every response on the first two pages before completing the portal so I need the pop up to always appear if any box on the first two pages isn't ticked. 

https://user-images.githubusercontent.com/110963329/204940389-cbca66b9-abc1-400b-9022-29c0e9e4bd20.mov

### PR DESCRIPTION
**Pull request by DoxasticFox**
Closes #105.

### CODE REVIEW

File: app/javascript/consent-form.vue [hunk 1]
```diff
@@ -307,7 +307,10 @@ export default {
       if (!await this.showSpinner(this.saveAnswers)) {
         return;
       }
-      if ( this.modalFallback && this.checkboxAgreement.answer == 'no') {
+
+      const doesAgreeToAll = this.checkboxAgreementQuestions.every(
```
Comment (ammazzaw) [inline app/javascript/consent-form.vue:311]: Lol! This code looked so clear to me... I got distracted and it took me moment to remember I was looking at JS and not Ruby 😂

File: app/javascript/consent-form.vue [hunk 2]
```diff
@@ -389,9 +392,10 @@ export default {
     questionGroups() {
       return this.currentSurveyStep.groups
     },
-    checkboxAgreement() {
-      // TO DO: Change to include all question groups, not just the first
-      return this.answers[0].find(a => a.question_type == 'checkbox agreement')
+    checkboxAgreementQuestions() {
+      return this.answers.flatMap(
```
Comment (ammazzaw) [inline app/javascript/consent-form.vue:396]: Nice. For future reference `flatMap` calls just make me happy 🤷‍♂️ In fact, all functions that are just folds on iterables brighten my day 😉

File: features/consent/modal.feature [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+  Scenario: User sees the modal fallback when some checkbox agreement elements aren't checked
+    Given I do not exist as a user
+    And A study code exists
+    When I click on Register
+    And I fill in the user details
+    Then I should see the step one of consent
+    When I click on Next
+    Then I should see the step two of the consent
+    When I click all the checkboxes
```
Comment (ammazzaw) [inline features/consent/modal.feature:13]: Just making sure I understand. Here you're first selecting all check boxes and then, with one more click, de-selecting the last one to trigger the fallback popup, right? Makes sense 👍

**Review by ammazzaw — APPROVED** [review]

File: features/consent/modal.feature (hunk 3 above)
Comment (DoxasticFox) [reply features/consent/modal.feature:13]: Yep

**Pull request merged**